streaming/src/main/scala/org/apache/spark/streaming/Checkpoint.scala

```diff
@@ -102,7 +102,7 @@ class Checkpoint(ssc: StreamingContext, val checkpointTime: Time)
 private[streaming]
 object Checkpoint extends Logging {
   val PREFIX = "checkpoint-"
-  val REGEX = (PREFIX + """([\d]+)([\w\.]*)""").r
+  val REGEX = (PREFIX + """([\d]{9,})([\w\.]*)""").r
```

Comment:
I think it will technically introduce a behaviour change since it targets to support the `checkpoint-` name with numbers. Let's clarify it.

Reply:
The intention behind this change was to skip invalid (or perhaps too old) checkpoint files since numeric part of checkpoint file name consists of current time in milliseconds and therefore cannot be shorter than 9 digits.
This caused some unit tests to fail because they are using ManualClock which reports fake time allowing generation of shorter checkpoint file names like `checkpoint-2000` (where 2000 is supposedly current time in milliseconds).
Now I consider this change in regex redundant and unnecessary since filtering out directories and matching only the final component of a path (p.getName) should be sufficient to prevent MatchErrors.
I will revert the change this change to fix unit test fails.